accord-core/src/main/java/accord/utils/async/AsyncChains.java

```diff
@@ -331,6 +331,62 @@ public static <V> AsyncChain<V> failure(Throwable failure)
         return new Immediate<>(failure);
     }
 
+    public static <V, T> AsyncChain<T> map(AsyncChain<V> chain, Function<? super V, ? extends T> mapper, Executor executor)
+    {
+        return chain.flatMap(v -> new Head<T>()
+        {
+            @Override
+            protected void start(BiConsumer<? super T, Throwable> callback)
+            {
+                try
+                {
+                    executor.execute(() -> {
+                        try
+                        {
+                            callback.accept(mapper.apply(v), null);
```

Comment:
I believe this is a potential bug, as we generally impose _precisely once_ semantics on our callbacks. If the `mapper.apply` fails we should invoke `callback.accept(null, t)` however if `callback.accept(v, null)` fails we should not, as the callback is now responsible for the outcome.

Reply:
fixed